src/axl.c

```diff
@@ -309,6 +310,92 @@ int AXL_Finalize (void)
     return rc;
 }
 
+
+/** Set a AXL config parameters */
+kvtree* AXL_Config(const int id, const kvtree* config)
```

Comment:
I would get rid of the `id` arg, and make the per-ID values subtrees within the main tree.  For example:

```C
    kvtree *tree;
    kvtree *id;

    tree = kvtree_new();

    /* Add some config for AXL ID 1 */
    id = kvtree_set_kv_int(tree, "id", 1);
    kvtree_set_kv_int(id, "compression", 1);
    kvtree_set_kv_int(id, "file_buf_size", 65536);

    /* Add some config for AXL ID 42 */
    id = kvtree_set_kv_int(tree, "id", 42);
    kvtree_set_kv_int(id, "file_buf_size", 1048576);

    printf("Whole tree ---\n");
    kvtree_print(tree, 4);

    /* Lookup an id 1's values */
    id = kvtree_get_kv_int(tree, "id", 1);
    printf("Just ID 1 ---\n");
    kvtree_print(id, 4);
```
```
Whole tree ---
      id
        42
          file_buf_size
            1048576
        1
          file_buf_size
            65536
          compression
            1
Just ID 1 ---
      file_buf_size
        65536
      compression
        1
```

Reply:
Uh, I have a not quite pull request (but a branch) that actually implements per transfer options. No thread locking yet. The diff can be seen here:

https://github.com/rhaas80/AXL/compare/rhaas/keyvalue_get...rhaas80:rhaas/per_transfer

and the branch itself is

https://github.com/rhaas80/AXL/tree/rhaas/per_transfer

and the second last commit (https://github.com/rhaas80/AXL/commit/b6a170077a576fa8cfca1c42eba4623449848d1c) is the one that goes from `id` being an argument to encoding it in the kvtree. I am not sure I *like* that solution, but it is certainly doable and the commit nicely shows the differences this implies to taking an `int id` argument in `AXL_Config`.

Reply:
in call on 2020-10-07:
* go with `kvtree` only argument test
* combine all 3 pull requests for `set`, `get` and `per_transfer` into single pull request
* might keep commits separate